03-auto_tick.xsh

```diff
@@ -2,303 +2,90 @@
-    body = ('This PR was created by the [cf-regro-autotick-bot](https://github.com/regro/cf-scripts).\n\n'
-            'The **cf-regro-autotick-bot** is a service to automatically track the dependency graph '
-            'and propose package version updates for conda-forge. It is very likely that the current ' 
-            'package version for this feedstock is out of date.\n\n'
-            'Notes and instructions for merging this PR:\n'
-            '1. Please check that the dependencies have not changed. \n'
-            '2. Please merge the PR only after the tests have passed. \n'
-            "3. Feel free to push to the bot's branch to update this PR if needed. \n"
-            "4. The bot will almost always only open one PR per version. \n\n"
-            "If you would like a local version of this bot, you might consider using "
-            "[rever](https://regro.github.io/rever-docs/). "
-            "Rever is a tool for automating software releases and forms the "
-            "backbone of the bot's conda-forge PRing capability. Rever is both "
-            "conda (`conda install -c conda-forge rever`) and pip "
-            "(`pip install re-ver`) installable.\n\n"
-            'Finally, feel free to drop us a line if there are any '
-            '[issues](https://github.com/regro/cf-scripts/issues)! ')
-    # Statement here
-    template = '|{name}|{new_version}|[![Anaconda-Server Badge](https://img.shields.io/conda/vn/conda-forge/{name}.svg)](https://anaconda.org/conda-forge/{name})|\n'
-    if len(pred) > 0:
-        body += ('\n\nHere is a list of all the pending dependencies (and their '
-                 'versions) for this repo. '
-                 'Please double check all dependencies before merging.\n\n')
-        # Only add the header row if we have content. Otherwise the rendered table in the github comment
-        # is empty which is confusing
-        body += '''| Name | Upstream Version | Current Version |\n|:----:|:----------------:|:---------------:|\n'''
-    for p in pred:
-        body += template.format(name=p[0], new_version=p[1])
-    pr = repo.create_pull(title, 'master', head, body=body)
-    if pr is None:
-        print('Failed to create pull request!')
-    else:
-        print('Pull request created at ' + pr.html_url)
+    # push up
+    push_repo(feedstock_dir, migrator)
     print('Removing feedstock dir')
     rm -rf @(feedstock_dir)
     return True
 
 # gx = nx.read_yaml('graph.yml')
 gx = nx.read_gpickle('graph.pkl')
-gx2 = copy.deepcopy(gx)
-
-# Prune graph to only things that need builds
-for node, attrs in gx.node.items():
-    if not attrs['new_version']:
-        gx2.remove_node(node)
-    elif parse_version(str(attrs['new_version'])) <= parse_version(str(attrs['version'])):
-        gx2.remove_node(node)
-    if attrs.get('archived', False):
-        gx2.remove_node(node)
+# TODO: this needs to be sourced from somewhere better (top level?)
+migrators = []
```

Comment:
Maybe this could be an env var? Or maybe we should have a separate rc file that the scripts load?

Reply:
I was thinking on a `bot.xsh` that we source (since we have used that pattern a bunch). I expect wherever we put this it will get a bunch of traffic as migrations start and finish.

Reply:
Yep, that works, that is the pattern I was thinking of as an rc file.  You should probably add that to this PR, since this is where you need it.